code/modules/hydroponics/seeds.dm

```diff
@@ -20,6 +20,33 @@
 	GLOB.seed_list -= src
 	return ..()
 
+/obj/item/seeds/verb/rename_seeds() //set amount_per_transfer_from_this
+	set name = "Rename Seeds"
+	set category = "Object"
+	set src in usr
+	if(!ishuman(usr))
+		return
+	var/mob/living/carbon/human/user = usr
+	var/obj/item/seeds/S = user.get_active_hand()
+	if(!istype(S))
+		return
+	if (seed.roundstart)
+		to_chat(usr, SPAN_WARNING("You can't rename basic seeds!"))
+		return
+	var/new_name = copytext(reject_bad_text(input(user,"Rename seed variety?", "Set new seed variety name", "")), 1, MAX_NAME_LEN)
```

Comment:
tgui input here, probably

Reply:
I tried searching for TGUI text input, but I don't think it exists. There is a numerical one and a list one, but not string.

I checked character preferences for how that UI handles name changes and it's

`var/raw_name = input(user, "Choose your character's name:", "Character Preference")  as text|null`

so I don't think there is another way of handling it other than `input`?

Reply:
From Discord:
"harryob — Today at 12:00 AM
oh uhhh i still need to port that"

Reply:
Apparently tgui text input was added recently, so fixed it as requested.